src/delay.rs

```diff
@@ -62,24 +62,16 @@ impl DelayUs<u32> for Delay {
 
         // With c_ck up to 480e6, we need u64 for delays > 8.9s
 
-        #[cfg(all(feature = "singlecore", feature = "revision_v"))]
-        let mut total_rvr =
-            u64::from(us) * u64::from(self.clocks.c_ck().0 / 8_000_000);
-
-        // See errata ES0392 §2.2.3. Revision Y does not have the /8 divider
-        #[cfg(all(feature = "singlecore", not(feature = "revision_v")))]
-        let mut total_rvr =
-            u64::from(us) * u64::from(self.clocks.c_ck().0 / 1_000_000);
-
-        // Dual core has an additional divide by 8
-        #[cfg(all(feature = "dualcore", feature = "cm7"))]
-        let mut total_rvr =
-            u64::from(us) * u64::from(self.clocks.c_ck().0 / 8_000_000);
-
-        // CM4 dervived from HCLK
-        #[cfg(all(feature = "dualcore", feature = "cm4"))]
-        let mut total_rvr =
-            u64::from(us) * u64::from(self.clocks.hclk().0 / 8_000_000);
+        let mut total_rvr = if cfg!(not(feature = "revision_v")) {
+            // See errata ES0392 §2.2.3. Revision Y does not have the /8 divider
```

Comment:
Unrelated side note, how does stuff like this happen? Do they just forget their divider?

Reply:
I'd guess that design components are simulated/verified separately to reduce the complexity, with functional models standing-in for the rest of the chip in each simulation. And this was just missed at some point, then only noticed again when running code on real silicon. Interesting that it took until the second spin "Y" before this was found, so likely the first spin "Z" wasn't even capable of running all their bring-up code (or it wasn't before the deadline to tape-out the second spin).